ignite/contrib/handlers/throughput_benchmark_wrapper.py

```diff
@@ -0,0 +1,216 @@
+import torch
+import torch.jit
+import contextlib
+from typing import Callable, Union
+from ignite.engine import Events, Engine
+from torch.utils import ThroughputBenchmark
+from torch.utils.throughput_benchmark import ExecutionStats  # for linting
+
+
+# TODO: Discuss device implications
+class ThroughputBenchmarkWrapper:
+    """
+    This class is wrapper around the ThroughputBenchmark wrapper from torch.
+    ThroughputBenchmark is responsible for executing a PyTorch
+    module (nn.Module or ScriptModule)
+    under an inference server like load. It can emulate multiple calling threads
+    to a single module provided.
+
+    Please note that even though nn.Module is supported, it might incur an overhead
+    from the need to hold GIL every time we execute Python code or pass around
+    inputs as Python objects. As soon as you have a ScriptModule version of your
+    model for inference deployment it is better to switch to using it in this
+    benchmark.
+
+    Args:
+        model (Union[torch.nn.Module, torch.jit.ScriptModule]): model which will
+            be used to run the benchmark on. Please keep in mind that most reliable
+            test results will be achieved with a `ScriptModule`.
+        num_calling_threads (int, optional): Number of threads that will call the model.
+            Defaults to 1.
+        num_warmup_iters (int, optional): Warmup iters are used to make sure we run a module
+            a few times before actually measuring things. This way we avoid cold
+            caches and any other similar problems. This is the number of warmup
+            iterations for each of the threads in separate.
+            Defaults to 10.
+        num_iters (int, optional): Number of iterations the benchmark should run with.
+            This number is separate from the warmup iterations. Also the number is
+            shared across all the threads. Once the num_iters iterations across all
+            the threads is reached, we will stop execution. Though total number of
+            iterations might be slightly larger. Which is reported as
+            stats.num_iters where stats is the result of this function.
+            Defaults to 100.
+
+    Examples:
+
+        .. code-block:: python
+
+            from ignite.contrib.handlers import ThroughputBenchmarkWrapper
+
+            # dataloader gets batches of (X, y) with many samples per batch
+            dataloader = ...
+            evaluator = ...
+            model = ...
+            throughput_benchmark = ThroughputBenchmarkWrapper(model)
+
+            # We decide that 10 batches of the datasampler are enough different samples for benchmark
+            max_batches = 10
+            # batch are of type (X, y) -> default input_transform is can be used
+            with throughput_benchmark.attach(evaluator, max_batches=max_batches) as evaluator_with_benchmark:
+                evaluator_with_benchmark.run(dataloader)
```

Comment:
Correct me, if I'm wrong, but in this example you would run the whole evaluation once, wouldn't you? shouldn't we stop after `max_batches`?

Reply:
Yes, you are correct.
My idea was that the user could include all of the desired event handlers to the evaluator and then run it once including with the benchmark. It will log the batches until max_batches and then run the benchmark once it is completed.

But I am happy to use an early stopping procedure if you think the benchmark should be used more in a *stand-alone* fashion.

Reply:
Added an early stopping procedure if `max_batches` is reached and rewrote the documentation.